src/Compilers/CSharp/Portable/Compilation/AwaitExpressionInfo.cs

```diff
@@ -37,15 +31,12 @@ public override bool Equals(object obj)
 
         public bool Equals(AwaitExpressionInfo other)
         {
-            return object.Equals(this.GetAwaiterMethod, other.GetAwaiterMethod)
-                && object.Equals(this.IsCompletedProperty, other.IsCompletedProperty)
-                && object.Equals(this.GetResultMethod, other.GetResultMethod)
-                && this.IsDynamic == other.IsDynamic;
+            return _awaitableInfo.Equals(other);
```

Comment:
>_awaitableInfo.Equals(other); [](start = 19, length = 29)

In addition, right now equality of AwaitableInfo take an additional piece of information into account, the AwaitableInfo.Type. Which means that, if we simply rely on it, we are likely to change behavior, i.e. equal instances of AwaitExpressionInfo are not going to equal after this change.